lib/Mojolicious/Controller.pm

```diff
@@ -272,6 +272,11 @@ sub url_for {
   return $url;
 }
 
+sub url_for_asset {
+  my ($self, $asset) = @_;
+  return $self->url_for($self->app->static->asset_path($asset));
+}
```

Comment:
I'm surprised this is a method and not a helper.

Reply:
Same reason `url_for` is a method, it's too essential to encourage overloading. Too many things will depend on it working a certain way.

Reply:
One good example is plugins adding their own public directory to the static file server and then depending on the default behaviour of `url_for_asset`.